content-src/components/DiscoveryStreamComponents/List/List.jsx

```diff
@@ -1,15 +1,65 @@
 import {connect} from "react-redux";
 import React from "react";
 
-export class _List extends React.PureComponent {
-  render() {
-    // const feed = this.props.DiscoveryStream.feeds[this.props.feed.url];
-    return (
-      <div className="ds-list">
-        List
+// XXX this arbitrarily chops the text when using a screenreader, unlike
+// text-overflow: ellipsis.  At some point, we should do better.
+function clampText(string, length) {
+  if (string.length <= length) {
+    return string;
+  }
+
+  /* eslint-disable-next-line prefer-template */
+  return string.substring(0, length - 1) + "…";
+}
+
+// XXX ellipsis
+function Recommendation(props) {
+  // XXX performance: get feeds to send appropriately size images rather
+  // than waiting longer and scaling down on client?
+  return (
+    <li className="ds-list-item">
+      <div className="ds-list-item-text">
+        <div className="ds-list-item-link-wrapper">
+            <a className="ds-list-item-link" href={props.url}>
+              <b>
+                {clampText(props.title, 60)}
+              </b>
+            </a>
+          </div>
+        <div className="ds-list-item-info">
+          <span>{`${props.domain} · TODO:Topic`}</span>
+        </div>
       </div>
-    );
+      <img className="ds-list-image" src={props.image_src} />
+    </li>
+  );
+}
+
+function _List(props) {
+  const feed = props.DiscoveryStream.feeds[props.feed.url];
+
+  let recs;
+  if (feed) {
+    recs = feed.data.recommendations;
+
+    // XXX
+    /* eslint-disable-next-line no-console */
+    console.log(feed);
   }
+
+  if (!recs) {
+    return null;
+  }
+
+  // XXX does rec.id meet necessary constraints for `key`?
```

Comment:
Should be fine. If it doesn't it probably means we have dupes in the data which would be more bad than a duplicate id.

Reply:
If you're worried, sometimes I grab the index as well and do `${id}-${index}`